src/transformers/pipelines/fill_mask.py

```diff
@@ -74,8 +83,23 @@ def __init__(
 
         self.check_model_type(TF_MODEL_WITH_LM_HEAD_MAPPING if self.framework == "tf" else MODEL_FOR_MASKED_LM_MAPPING)
         self.top_k = top_k
+        self.targets = targets
+        if self.tokenizer.mask_token_id is None:
+            raise PipelineException(
+                "fill-mask", self.model.base_model_prefix, "The tokenizer does not define a `mask_token`."
+            )
 
-    def ensure_exactly_one_mask_token(self, masked_index: np.ndarray):
+    def get_masked_index(self, input_ids: GenericTensor) -> np.ndarray:
+        if self.framework == "tf":
+            masked_index = tf.where(input_ids == self.tokenizer.mask_token_id).numpy()
+        elif self.framework == "pt":
+            masked_index = torch.nonzero(input_ids == self.tokenizer.mask_token_id, as_tuple=False)
+        else:
+            raise ValueError("Unsupported framework")
+        return masked_index
+
+    def _ensure_exactly_one_mask_token(self, input_ids: GenericTensor) -> np.ndarray:
+        masked_index = self.get_masked_index(input_ids)
         numel = np.prod(masked_index.shape)
```

Comment:
This will return two when an array of sequences was passed, so it will never pass in that case. See:

```py
>>> masker(["Cool [MASK] !"])
[...]
    raise PipelineException(
transformers.pipelines.base.PipelineException: More than one mask_token ([MASK]) is not supported
```

Reply:
We should never pass a 2d array, do you mind expanding your test case, I was not able to reproduce ?

Reply:
I was able to reproduce by adding an explicit test for tokenizer with no pad token.

I added it to the test suite and fixed everything, now everyhing should work.
The end result is not really pretty, I intend to solve most uglyness after the tests are passed if possible.